src/utilities/NodeEndpointUtilities.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 NEM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { WebSocketConfig } from '../infrastructure/Listener';
+import { NetworkTypes } from '../models/node/NetworkTypes';
+import { NEMLibrary } from '../NEMLibrary';
+
+export const nodeEndpoints = (): Array<WebSocketConfig> => {
+  if (NEMLibrary.getNetworkType() === NetworkTypes.MAIN_NET) {
```

Comment:
I'm not sure i understand why this function was created? What does this do that the code didn't have before?

Reply:
i was requiring this to be passed in to connect to a node, there are issues automatically connecting to the list of nodes, its def temporary.  Remember you asked why you have to pass this in to the observer, this is that